### PR TITLE
Add checking required field for exporting iOS

### PR DESCRIPTION
**Pull request by volzhs**
without specifying Identifier, it will cause `can not be parsed` error on Xcode

### CODE REVIEW

**Comment by akien-mga** [comment]
Related to #23172.

Is any non-empty string valid, or should it do some more extensive validation like in #23320?

**Comment by volzhs** [comment]
right, it's better to validate format like #23320 did.

**Comment by volzhs** [comment]
added checking validation

File: platform/iphone/export/export.cpp [hunk 1]
```diff
@@ -99,6 +99,70 @@ class EditorExportPlatformIOS : public EditorExportPlatform {
...
+				return false;
+			}
+			if (c == '.') {
+				segments++;
+				first = true;
+				continue;
+			}
+			if (!((c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9') || c == '_')) {
+				if (r_error) {
+					*r_error = "The character '" + String::chr(c) + "' is not allowed in Android application package names.";
```
Comment (samgreen) [inline platform/iphone/export/export.cpp]: Android?

File: platform/iphone/export/export.cpp [hunk 2]
```diff
@@ -987,6 +1051,14 @@ bool EditorExportPlatformIOS::can_export(const Ref<EditorExportPreset> &p_preset
 		err += "App Store Team ID not specified - cannot configure the project.\n";
 	}
 
+	String identifier = p_preset->get("application/identifier");
+	String pn_err;
+	if (identifier.length() == 0) {
```
Comment (samgreen) [inline platform/iphone/export/export.cpp]: Is this duplicating the check in is_package_name_valid?

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Now that Godot 3.1 has been released, we only plan to cherry-pick critical bug fixes to the 3.0 branch, unless there is very strong support in favor of a specific non-critical issue. Removing cherrypick label for 3.0.

**Pull request merged**